app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -6,6 +6,7 @@ def make_offer
       decision = MakeAnOffer.new(
         application_choice: application_choice,
         offer_conditions: params.dig(:data, :conditions),
+        offer_course_option_id: params.dig(:data, :offer_course_option_id),
```

Comment:
What would you think of making the API accept a Course object instead of the `course_option_id`?

https://qa.apply-for-teacher-training.education.gov.uk/api-docs/reference#course-object

I imagine the API looking something like this:

```rb
describe 'making an offer for another course' do
  it 'returns the updated application' do
    application_choice = create_application_choice_for_currently_authenticated_provider(
      status: 'awaiting_provider_decision',
    )

    other_course_option = course_option_for_provider(provider: currently_authenticated_provider)

    post_api_request '/api/v1/applications/#{application_choice.id}/offer', params: {
      'data' => {
        'conditions' => [],
        'course' => {
          'provider_code' => other_course_option.course.provider.code,
          'course_code' => other_course_option.course.code,
          'site_code' => other_course_option.site.code,
        }
      },
    }

    expect(parsed_response['data']['attributes']['offer']).to eq(
      'conditions' => [],
      'course' => {
        'provider_code' => other_course_option.course.provider.code,
        'course_code' => other_course_option.course.code,
        'site_code' => other_course_option.site.code,
      },
    )
  end
end
```

Reply:
(What Duncan said)